### PR TITLE
Add issuer to auth

### PR DESCRIPTION
**Pull request by kw217**
After discussion re #14 we agreed to add `issuer` - knowing who did something is very important for audit purposes; just knowing who it was done to is not enough. Also it is useful to be able to report per-issuer (i.e., per-third-party-app) statistics from server applications, and this is easier if that information is present within the application (rather than just in logs).

We agreed to drop `auth_type` and `expiry_deadline`. The first (`auth_type`) would encourage services to have auth-type-dependent behaviour, making the testing matrix more complex; the second (`expiry_deadline`) is not actually useful - authenticators may wish to do internal caching based on deadlines, but the validity period of the auth data doesn't imply anything about the validity period of the requester's authorization.

This is a breaking change (new field in struct), hence there is a version bump. If approved, I'll tag and release the library to cargo.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+### Added
+
+### Changed
+- Authorization struct now has new field `issuer`.
+
+## [0.5.0] - 2017-09-18
+- Start of changelog.
+
+[Unreleased]: https://github.com/olivierlacan/keep-a-changelog/compare/0.5.0...HEAD
+[0.5.0]: https://github.com/olivierlacan/keep-a-changelog/compare/0.4.0...0.5.0
```
Comment (mirw) [inline CHANGELOG.md]: ?

File: src/auth.rs [hunk 2]
```diff
@@ -17,10 +17,16 @@ pub enum Scopes {
...
+    /// Subject for which authorization is granted
+    /// (i.e., what may be accessed.)
     pub subject: String,
-    /// Authorization scopes of the authenticated subject
+    /// Scopes for which authorization is granted
+    /// (i.e., what types of access are permitted).
     pub scopes: Scopes,
+    /// Authenticated identity of the party to which
+    /// authorization has been granted, if available
+    /// (i.e., who is doing the accessing).
```
Comment (mirw) [inline src/auth.rs]: Is "who is doing the accessing" actually correct here (or, at least, is it over simplifying)?  If I make a request to microservice X to access microservice Y to retrieve data from store Z, you could say that X, Y and Z are all doing the accessing, but I think the issuer would still be me.  Correct?  Maybe it's better to either drop the simplified explanation, or flesh it out?

File: CHANGELOG.md (hunk 1 above)
Comment (kw217) [reply CHANGELOG.md]: D'oh, thanks.

File: src/auth.rs (hunk 2 above)
Comment (kw217) [reply src/auth.rs]: I've fleshed it out a bit, with reference to RFC6749 and RFC7519. Hopefully this is better?

**Comment by mirw** [comment]
Looks good.

**Pull request merged**